core/repository/sparql/src/main/java/org/eclipse/rdf4j/repository/sparql/SPARQLConnection.java

```diff
@@ -391,11 +392,13 @@ public void commit() throws RepositoryException {
 		synchronized (transactionLock) {
 			if (isActive()) {
 				synchronized (transactionLock) {
-					SPARQLUpdate transaction = new SPARQLUpdate(client, null, sparqlTransaction.toString());
-					try {
-						transaction.execute();
-					} catch (UpdateExecutionException e) {
-						throw new RepositoryException("error executing transaction", e);
+					if (sparqlTransaction.length() > 0) {
```

Comment:
Would be nice if you could add a comment here. 

// empty transactions commit as no-op, some SPARQL endpoints don't handle empty transactions